Smt/String.lean

```diff
@@ -10,24 +10,45 @@ import Smt.Transformer
 
 namespace Smt.String
 
-open Lean
-open Lean.Expr
+open Lean Expr
 open Smt.Transformer
 
-@[Smt] def transformStringGetOp : Transformer
+@[Smt] def replaceConst : Transformer
+  | const `Char.ofNat ..     => pure none
+  | const `String.Pos.mk ..  => pure none
+  | const `String.replace .. => pure (mkConst `str.replace_all)
+  | const `String.length ..  => pure (mkConst `str.len)
+  | const `String.append ..  => pure (mkConst (Name.mkSimple "str.++"))
+  | e                        => pure e
+
+@[Smt] def removeStringPosMk : Transformer
+  | app (const `String.Pos.mk ..) e _ => applyTransformations e
```

Comment:
Why not `| const `String.Pos.mk .. => pure none` ?

Reply:
Oh wait, you already have that above. Why is this one needed?

Reply:
It shouldn't be there. I just forgot to remove it.